### ISSUE TITLE
Call to flexVersion() causes an NPE on a new repo

### ISSUE THREAD

**Issue by asvoboda** [palantir/gradle-flexversion#34]
Repro with the following steps:

```
$ mkdir testing
$ cd testing
$ git init .
$ git symbolic-ref HEAD
refs/heads/master
$ git rev-parse refs/heads/master
refs/heads/master
fatal: ambiguous argument 'refs/heads/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

When calling flexVersion() from within the build.gradle, the following error message is printed:

```
Caused by: java.lang.NullPointerException
    at org.eclipse.jgit.lib.ObjectIdOwnerMap.get(ObjectIdOwnerMap.java:131)
    at org.eclipse.jgit.revwalk.RevWalk.parseAny(RevWalk.java:837)
    at org.eclipse.jgit.revwalk.RevWalk.parseCommit(RevWalk.java:752)
    at org.eclipse.jgit.revwalk.RevWalk$parseCommit.call(Unknown Source)
    at com.palantir.gradle.versions.flexversioning.GitUtils.getCommitHead(GitUtils.groovy:59)
    at com.palantir.gradle.versions.flexversioning.GitUtils$getCommitHead$0.call(Unknown Source)
    at com.palantir.gradle.versions.flexversioning.FlexVersionPlugin.buildFlexVersion(FlexVersionPlugin.groovy:48)
    at com.palantir.gradle.versions.flexversioning.FlexVersionPlugin$buildFlexVersion.call(Unknown Source)
    at com.palantir.gradle.versions.flexversioning.FlexVersionConvention.flexVersion(FlexVersionConvention.groovy:33)
    at com.palantir.gradle.versions.flexversioning.FlexVersionConvention.invokeMethod(FlexVersionConvention.groovy)
    at org.gradle.api.internal.BeanDynamicObject$GroovyObjectAdapter.invokeMethod(BeanDynamicObject.java:275)
    at org.gradle.api.internal.BeanDynamicObject.invokeMethod(BeanDynamicObject.java:127)
    at org.gradle.api.internal.plugins.DefaultConvention$ExtensionsDynamicObject.invokeMethod(DefaultConvention.java:212)
    at org.gradle.api.internal.CompositeDynamicObject.invokeMethod(CompositeDynamicObject.java:150)
    at org.gradle.groovy.scripts.BasicScript.methodMissing(BasicScript.java:79)
    at build_1oq5i4ib64lcueolghjn36fe8$_run_closure4.doCall(/Users/asvoboda/dev/gotham-containers-cli/build.gradle:46)
    at build_1oq5i4ib64lcueolghjn36fe8$_run_closure4.doCall(/Users/asvoboda/dev/gotham-containers-cli/build.gradle)
    at org.gradle.api.internal.plugins.DefaultExtraPropertiesExtension.methodMissing(DefaultExtraPropertiesExtension.java:75)
    at build_1oq5i4ib64lcueolghjn36fe8.run(/Users/asvoboda/dev/gotham-containers-cli/build.gradle:75)
    at org.gradle.groovy.scripts.internal.DefaultScriptRunnerFactory$ScriptRunnerImpl.run(DefaultScriptRunnerFactory.java:91)
    ... 50 more
```

We should print out a better error message rather than throwing an NPE.

**Comment by ash211** [comment]
For those seeing this problem in the future, you can work around it by creating a first commit in your repo and then flexVersion() should work.

**Comment by kostasniktas** [comment]
I thought we had made a better error message for this but I guess it was for no git repo:
https://github.com/palantir/gradle-flexversion/commit/f3b6dabb73a2b82568c1bcd710e8333a13e1c8c8

We can do something similar.  Either throw a clean error message or maybe just set the count to 0

**Comment by kostasniktas** [comment]
Fixed in f489b35deea7f30abd3839fbceca8c6414bf6868
